### ISSUE TITLE
BUG: `deepcopy(Generator())` does not copy the `SeedSequence` state

### ISSUE THREAD

**Issue by lucianopaz** [numpy/numpy#27727]
### Describe the issue:

The standard library's `copy.deepcopy` of a random `Generator` object copies the `BitGenerator` state, but not the `SeedSequence` state.

I'm not sure if this behavior is intentional or not. It's easy to create a new `Generator` object that has new `BitGenerator` and `SeedSequence` object's that match the same state than the original object by doing this:

```python
def copy_rng(rng: np.random.Generator) -> np.random.Generator:
    seed_seq = np.random.SeedSequence(**rng.bit_generator.seed_seq.state)
    bit_generator_class = getattr(np.random, rng.bit_generator.state["bit_generator"])
    bit_generator = bit_generator_class(seed_seq)
    bit_generator.state = rng.bit_generator.state
    return np.random.Generator(bit_generator)
```

### Reproduce the code example:

```python
import numpy as np
from copy import deepcopy

rng = np.random.default_rng(12345)
rng2 = deepcopy(rng)

assert rng.bit_generator.state == rng2.bit_generator.state  # Works
assert rng.bit_generator.seed_seq.state == rng2.bit_generator.seed_seq.state  # Fails
```

### Python and NumPy Versions:

numpy.version = 1.26.4
sys.version = 3.11.10 | packaged by conda-forge | (main, Oct 16 2024, 01:26:25) [Clang 17.0.6 ]

### Runtime Environment:

[{'numpy_version': '1.26.4',
  'python': '3.11.10 | packaged by conda-forge | (main, Oct 16 2024, 01:26:25) '
            '[Clang 17.0.6 ]',
  'uname': uname_result(system='Darwin', node='Mac.homenet.telecomitalia.it', release='24.1.0', version='Darwin Kernel Version 24.1.0: Thu Oct 10 21:02:26 PDT 2024; root:xnu-11215.41.3~2/RELEASE_ARM64_T8122', machine='arm64')},
 {'simd_extensions': {'baseline': ['NEON', 'NEON_FP16', 'NEON_VFPV4', 'ASIMD'],
                      'found': ['ASIMDHP'],
                      'not_found': ['ASIMDFHM']}}]

### Context for the issue:

I'm working on MCMC algorithms in `pymc` and wanted to ensure reproducibility. In that context, some generators `spawn` others and I expected that `deepcopy` would also copy the `SeedSequence` state between generators. I caught this behavior in some of my tests and I thought that if it was designed to be like this, it was not intuitive. I have a workaround in place to copy generators that copy both the state and `seed_seq`, but I think that it would be worth it for `numpy` to address this problem upstream.

**Comment by seberg** [comment]
This is fixed/dosn't reproduce in NumPy 2.0.0.  So there is probably nothing more to do, since there isn't anothr 1.x release planned I think.

**Comment by lucianopaz** [comment]
Thanks @seberg
